pkg/dns/aws/dns.go

```diff
@@ -208,48 +195,41 @@ func NewProvider(config Config, operatorReleaseVersion string) (*Provider, error
 		config:    config,
 		idsToTags: map[string]map[string]string{},
 		lbZones:   map[string]string{},
-	}, nil
+	}
+	if err := validateServiceEndpoints(p); err != nil {
+		return nil, fmt.Errorf("failed to validate aws provider service endpoints: %v", err)
+	}
+	return p, nil
 }
 
-// urlContainsValidRegion checks whether uri is valid based on provided region.
-func urlContainsValidRegion(region, uri string) bool {
-	switch {
-	case strings.Contains(uri, Route53Service):
-		switch region {
-		case endpoints.CnNorth1RegionID, endpoints.CnNorthwest1RegionID:
-			if uri == chinaRoute53Endpoint {
-				return true
-			}
-		case endpoints.UsGovEast1RegionID, endpoints.UsGovWest1RegionID:
-			if uri == govCloudRoute53Endpoint {
-				return true
-			}
-		default:
-			if uri == standardRoute53Endpoint || strings.Contains(uri, endpoints.UsEast1RegionID) {
-				return true
-			}
+// validateServiceEndpoints validates that provider clients can communicate
+// with associated API endpoints by each making a list/describe/get call.
+func validateServiceEndpoints(provider *Provider) error {
+	if provider.route53 != nil {
```

Comment:
Can `provider.route53`, `provider.elb`, `provider.elbv2`, or `provider.tags` ever be nil?

Reply:
Not the way it's being called from `NewProvider()` today. I'll update to remove the nil check.